lib/Dancer/Plugin/SimpleCRUD.pm

```diff
@@ -610,10 +610,27 @@ CONFIRMDELETE
             my $dbh = database($args{db_connection_name});
             my $where = _get_where_filter_from_args(\%args);
             $where->{$key_column} = $id;
-            $dbh->quick_delete($table_name, $where)
+
+            my %params = params;
+            my $meta_for_hook = {
+                args => \%args,
+                params => \%params,
+                table_name => $table_name,
+                key_column => $key_column,
+            };
+            # fire the pre-delete hook, in case user wants to perform trickery before the delete
+            execute_hook('delete_row_pre_delete', $meta_for_hook );
```

Comment:
I thought about how you might want to allow the pre-delete hook to block the delete from continuing. But since the analogous add_edit_row_pre_save hook doesn't let you stop the save from continuing, I suggest the pre-delete hook act the same for now. 

We can revisit pre-*-hooks aborting saves/edits/adds in a separate PR.

Sound good? Also, Does "Changes approved" mean that this is ready for merge?